py-polars/polars/internals/dataframe/frame.py

```diff
@@ -4239,7 +4243,7 @@ def with_column(self, column: pli.Series | pli.Expr) -> DataFrame:
         │ 3   ┆ 4   ┆ 16.0      │
         │ 5   ┆ 6   ┆ 36.0      │
         └─────┴─────┴───────────┘
-        >>> df.with_column(pl.col("a") ** 2)  # replaced
+        >>> df.with_columns(pl.col("a") ** 2)  # replaced
```

Comment:
Can we add these doc examples to `with_columns`? I want it to be obvious that `with_columns` also works when passing a single Series / Expression. That page will probably get a lot of traffic after we release this change!

We can remove the examples here - if you want to keep them, I'd suggest using `with_column` rather than `with_columns`, because technically you're giving doc examples of an entirely different method.

Reply:
I have removed the examples in the `with_column` docstring, so it is even more clear we dont want to use this anymore. The warning in the docstring + the code warning make that clear now.

I have added, using the dataframe as used in `with_columns`, examples of passing in a single expression, either with a new name or an existing name. That was missing from `with_columns`, and is important for users to know.

Reply:
Great!